### PR TITLE
feat(dataarts): add process resource support

### PR DESCRIPTION
**Pull request by jinyangyang222**
**What this PR does / why we need it**:
add process resource support for dataarts architecture
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
add process resource support for dataarts architecture
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/dataarts" TESTARGS="-run TestAccResourceArchitectureProcess_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dataarts -v -run TestAccResourceArchitectureProcess_basic -timeout 360m -parallel 4
=== RUN   TestAccResourceArchitectureProcess_basic
=== PAUSE TestAccResourceArchitectureProcess_basic
=== CONT  TestAccResourceArchitectureProcess_basic
--- PASS: TestAccResourceArchitectureProcess_basic (18.29s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dataarts  18.322s

```

### CODE REVIEW

**Comment by Lance52259** [comment]
The document name is wrong, should be `dataarts_studio_catalog`, not `dataarts_sudio_catalog`

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_catalog.go [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+				Computed:    true,
+				Description: "The creator of the catalog.",
+			},
+			"updated_by": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The last editor of the catalog.",
+			},
+			"children": {
+				Type:        schema.TypeList,
```
Comment (Lance52259) [inline huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_catalog.go:116]: the response list is in order?
B, A, C -> A, B, C

File: docs/resources/dataarts_sudio_catalog.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+* `parent_id` - (Optional, String) Specifies the parent catalog ID of catalog.  
+  It's **Required** when you create a subordinate catalog, and directory level cannot exceed `3` layer.
+
+* `prev_id` - (Optional, String) Specifies the ID of the previous node in the catalog.  
+  When querying the catalog, if the field is null, it means that is the first node in the current catalog directory.
+
+* `next_id` - (Optional, String) Specify the ID of the next node in the catalog.  
+  When querying the catalog, if the field is null, it means that is the last node in the current catalog directory.
+
+-> **NOTE:** The `prev_id` and `next_id` will change with the position of the catalog in the current directory.
```
Comment (Lance52259) [inline docs/resources/dataarts_sudio_catalog.md]:
```suggestion
-> The `prev_id` and `next_id` will change with the position of the catalog in the current directory.
```
`->` and `-> **Note**:` are same.

`->` means note (with blue)
`~>` means note (with yellow)
`!>` means warning

Comment (jinyangyang222) [reply docs/resources/dataarts_sudio_catalog.md]: ok, fixed.

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_catalog.go [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+	}
+
+	paths := strings.Split(d.Get("qualified_id").(string), ".")
+	jsonPaths := fmt.Sprintf("data.value[?id=='%s']", paths[0])
+	for i, path := range paths {
+		if i == 0 {
+			continue
+		}
+		jsonPaths += fmt.Sprintf("[children][][?id=='%s'][]", path)
+	}
```
Comment (Lance52259) [inline huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_catalog.go]:
```suggestion
	for i, path := range paths[1:] {
		jsonPaths += fmt.Sprintf("[children][][?id=='%s'][]", path)
	}
```

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_catalog.go [hunk 4]
```diff
@@ -0,0 +1,410 @@
...
+
+	catalogs := utils.PathSearch(jsonPaths, getRespBody, make([]interface{}, 0)).([]interface{})
+	if len(catalogs) == 0 {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "DataArts Studio catalog")
+	}
+
+	catalog := catalogs[0]
+	d.SetId(utils.PathSearch("id", catalog, "").(string))
```
Comment (Lance52259) [inline huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_catalog.go]: how about take this code into the import method?

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_catalog.go (hunk 3 above)
Comment (jinyangyang222) [reply huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_catalog.go]: fixed.

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_catalog.go (hunk 4 above)
Comment (jinyangyang222) [reply huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_catalog.go]: fixed.

File: huaweicloud/provider.go [hunk 5]
```diff
@@ -1103,6 +1103,7 @@ func Provider() *schema.Provider {
 			"huaweicloud_dataarts_architecture_directory":       dataarts.ResourceArchitectureDirectory(),
 			"huaweicloud_dataarts_architecture_subject":         dataarts.ResourceArchitectureSubject(),
 			"huaweicloud_dataarts_architecture_business_metric": dataarts.ResourceBusinessMetric(),
+			"huaweicloud_dataarts_architecture_process":         dataarts.ResourceProcess(),
```
Comment (ShiChangkuo) [inline huaweicloud/provider.go]: suggest to add Architecture prefix
```suggestion
			"huaweicloud_dataarts_architecture_process":         dataarts.ResourceArchitectureProcess(),
```

Comment (jinyangyang222) [reply huaweicloud/provider.go]: fixed.

**Pull request merged**